src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralLetter.java

```diff
@@ -109,7 +108,9 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
 
         log.info("Caseworker create general letter about to submit callback invoked for Case Id: {}", details.getId());
 
-        Applicant applicant = getApplicantByParty(caseData);
+        Applicant applicant = GeneralParties.RESPONDENT.equals(caseData.getGeneralLetter().getGeneralLetterParties())
+                ? caseData.getApplicant2()
+                : caseData.getApplicant1();
```

Comment:
I don't think this takes in to account the "Other" option. The old CaseLetterPrinter had:

```
      GeneralParties parties = Optional.ofNullable(firstElement(caseData.getGeneralLetters()))
                .map(element -> element.getValue().getGeneralLetterParties())
                .orElse(GeneralParties.OTHER);

            var recipientName = switch (parties) {
                case RESPONDENT -> caseData.getApplicant2().getFullName();
                case APPLICANT -> caseData.getApplicant1().getFullName();
                case OTHER -> caseData.getGeneralLetter().getOtherRecipientName();
            };
```